app/plugins/basic-authentication.md

```diff
@@ -62,6 +62,8 @@ form parameter                             | default | description
 
 In order to use the plugin, you first need to create a consumer to associate one or more credentials to. The Consumer represents a developer using the final upstream service.
 
+A note about `hide_credentials`: Setting it to `false` (the default) implies that the upstream service requires basic authentication. Setting it to `true` puts auth into Kong. Use cURL commands directly to the upstream to determine how to set this value.
```

Comment:
Ditto, better remove this paragraph, it seems to only add confusion. Suddenly I am left wondering whether I should still have authentication in my upstream service, or not, or if Kong even solves this problem at all because this doc seems to suggest I should still have basic-auth upstream?

Additionally, the passing the credentials from Kong to the upstream service is unlikely to work, unless the upstream service shares Kong's Consumers and registered (or vice-versa). Really, this is just very confusing imho.

Reply:
I've removed it. However, the crux of what is confusing is that we allow hide_credentials to be set at all. It seems that the only reason to setup the plugin at all is if you do not have auth on the upstream. If you do (the hide_credentials=false setting) you shouldn't use the plugin in the first place. This also brings into question the default value, which given our discussion, should default to true.

Reply:
I definitely agree, `hide_credential` is a confusing (and maybe unnecessary) configuration value... Historically, authentication plugins would do nothing to strip away the credentials they matched (transparent proxy, etc...) until some users got confused or experienced strange behavior from their upstream trying to validate those Kong-specific credentials, and the possibility of stripping them away was added with this parameter, without changing the default to preserve backwards-compatibility (and because not much was given to it either, tbh). Nowadays it indeed seems like `true` should be the default albeit breaking the transparent proxy paradigm, but adhering to the API Gateway one.